lib/google/gax/operation.rb

```diff
@@ -155,7 +156,19 @@ def name
       # @return [Object, nil]
       #   The metadata of the operation. Can be nil.
       def metadata
-        return nil if @grpc_op.metadata.nil?
+        return if @grpc_op.metadata.nil?
+
+        @metadata_type ||= begin
+          lookup_uri = URI.parse(@grpc_op.metadata.type_url)
```

Comment:
Is this necessary? Will `grpc_op.metadata.type_name` get you the same result without the URI parsing? Ex:

```
require 'google/cloud/speech'

ctx = Google::Cloud::Speech::V1::SpeechContext.new
ctx.phrases << "a" << "b"

any = Google::Protobuf::Any.new
any.pack ctx

type = Google::Protobuf::DescriptorPool.generated_pool.lookup(any.type_name).msgclass
obj = any.unpack type
```

Reply:
I'll update this PR to use `#type_name`.